libavfilter/vf_vpp_qsv.c

```diff
@@ -460,11 +486,27 @@ static int config_output(AVFilterLink *outlink)
 #endif
     }
 
+    if (vpp->scaling_mode) {
+#ifdef QSV_HAVE_SCALING
+        memset(&vpp->scaling_conf, 0, sizeof(mfxExtVPPScaling));
```

Comment:
How about we define some macro for this?
```
#define INIT_VPP_PARAM(param, id) \
memset(param, 0, sizeof(*param)); \
param->Header...\
param.ext_buf[param.num_ext_buf++] = (mfxExtBuffer*)param;
```

Reply:
I may re-factor the code out, thanks for the comment.

Reply:
changed in the new version, thx.